src/BaseListView.php

```diff
@@ -4,248 +4,346 @@
 
 namespace Yiisoft\Yii\DataView;
 
-use InvalidArgumentException;
-use Yiisoft\Aliases\Aliases;
+use JsonException;
 use Yiisoft\Arrays\ArrayHelper;
-use Yiisoft\Data\Paginator\KeysetPaginator;
 use Yiisoft\Data\Paginator\OffsetPaginator;
-use Yiisoft\Data\Reader\CountableDataInterface;
-use Yiisoft\Data\Reader\DataReaderInterface;
-use Yiisoft\Data\Reader\FilterableDataInterface;
-use Yiisoft\Data\Reader\OffsetableDataInterface;
-use Yiisoft\Data\Reader\SortableDataInterface;
-use Yiisoft\Factory\Exceptions\InvalidConfigException;
+use Yiisoft\Data\Paginator\PaginatorInterface;
 use Yiisoft\Html\Html;
-use Yiisoft\View\WebView;
-use Yiisoft\Widget\Widget;
+use Yiisoft\Translator\TranslatorInterface;
+use Yiisoft\Yii\DataView\Exception\InvalidConfigException;
 use Yiisoft\Yii\DataView\Widget\LinkPager;
 use Yiisoft\Yii\DataView\Widget\LinkSorter;
 
+use function count;
+use function preg_replace_callback;
+
 /**
- * BaseListView is a base class for widgets displaying data from data provider
- * such as ListView and GridView.
+ * BaseListView is a base class for widgets displaying data from data provider such as ListView and GridView.
+ *
  * It provides features like sorting, paging and also filtering the data.
- * For more details and usage information on BaseListView, see the [guide article on data
- * widgets](guide:output-data-widgets).
+ *
+ * For more details and usage information on BaseListView:
+ * see the [guide article on data widgets](guide:output-data-widgets).
  */
 abstract class BaseListView extends Widget
 {
-    /**
-     * @var array the HTML attributes for the container tag of the list view.
-     * The "tag" element specifies the tag name of the container element and defaults to "div".
-     *
-     * @see Html::renderTagAttributes() for details on how attributes are being rendered.
-     */
+    public const BOOTSTRAP = 'bootstrap';
```

Comment:
How would it work with something else?

Reply:
The widgets need classes according to the css framework used, if I remove it the user will have to set everything to manually.

Reply:
1. What if the framework is neither bootstrap nor bulma? Can I set my own classes via widget factory config?
2. We can add examples of such configs to readme.
3. We can make two additional packages: `yii-dataview-bootstrap`, `yii-dataview-bulma`. Each package may provide its own fields config and require `yii-dataview`. This way user will be able to install the version needed.

Reply:
We can merge current variant for now but then we should create an issue to do it properly later.

Reply:
The widget is 100% configurable, for any framework, only that since we work with those two ccs frameworks, they are already configured by default and thus avoid having to do everything manually, but you can define each option for everything.

Reply:
OK. I still don't like that we have these two explicitly in the code though...

Reply:
I will gladly delete it, but it will have to be configured manually.